flink/v1.20/flink/src/main/java/org/apache/iceberg/flink/sink/CommittableToTableChangeConverter.java

```diff
@@ -0,0 +1,181 @@
+  public void open(OpenContext openContext) throws Exception {
+    super.open(openContext);
+    this.flinkJobId = getRuntimeContext().getJobId().toString();
+    if (!tableLoader.isOpen()) {
+      tableLoader.open();
+    }
+    this.table = tableLoader.loadTable();
+  }
+
+  @Override
+  public void snapshotState(FunctionSnapshotContext context) throws Exception {
+    manifestFilesToRemoveState.update(manifestFilesToRemoveList);
+  }
+
+  @Override
+  public void processElement(
+      CommittableMessage<IcebergCommittable> value,
+      ProcessFunction<CommittableMessage<IcebergCommittable>, TableChange>.Context ctx,
+      Collector<TableChange> out)
+      throws Exception {
+    if (value instanceof CommittableWithLineage) {
+      CommittableWithLineage<IcebergCommittable> committable =
+          (CommittableWithLineage<IcebergCommittable>) value;
+      TableChange tableChange = convertToTableChange(committable.getCommittable());
+      out.collect(tableChange);
+    }
+  }
+
+  private TableChange convertToTableChange(IcebergCommittable icebergCommittable)
+      throws IOException {
+    if (icebergCommittable == null || icebergCommittable.manifest().length == 0) {
+      return TableChange.empty();
+    }
+
+    DeltaManifests deltaManifests =
+        SimpleVersionedSerialization.readVersionAndDeSerialize(
+            DeltaManifestsSerializer.INSTANCE, icebergCommittable.manifest());
+    WriteResult writeResult =
+        FlinkManifestUtil.readCompletedFiles(deltaManifests, table.io(), table.specs());
+    manifestFilesToRemoveList.addAll(deltaManifests.manifests());
+
+    int dataFileCount = 0;
+    long dataFileSizeInBytes = 0L;
+    for (DataFile dataFile : writeResult.dataFiles()) {
+      dataFileCount++;
+      dataFileSizeInBytes += dataFile.fileSizeInBytes();
+    }
+
+    int posDeleteFileCount = 0;
+    long posDeleteRecordCount = 0L;
+    int eqDeleteFileCount = 0;
+    long eqDeleteRecordCount = 0L;
+
+    for (DeleteFile deleteFile : writeResult.deleteFiles()) {
+      switch (deleteFile.content()) {
+        case POSITION_DELETES:
+          posDeleteFileCount++;
+          posDeleteRecordCount += deleteFile.recordCount();
+          break;
+        case EQUALITY_DELETES:
+          eqDeleteFileCount++;
+          eqDeleteRecordCount += deleteFile.recordCount();
+          break;
+        default:
+          // Unexpected delete file types don't impact compaction task statistics, so ignore.
+          LOG.warn("Unexpected delete file content:{}", deleteFile.content());
+      }
+    }
+
+    TableChange tableChange =
+        TableChange.builder()
+            .dataFileCount(dataFileCount)
+            .dataFileSizeInBytes(dataFileSizeInBytes)
+            .posDeleteFileCount(posDeleteFileCount)
+            .posDeleteRecordCount(posDeleteRecordCount)
+            .eqDeleteRecordCount(eqDeleteRecordCount)
+            .eqDeleteFileCount(eqDeleteFileCount)
+            .commitCount(1)
+            .build();
```

Comment:
Instead of this could we add a `TableChange` constructor?
Like `TableChange(Iterable<DataFile> dataFiles,  Iterable<DeleteFile> deleteFiles)?

Maybe we could move the `CommittableToTableChangeConverter` to the maintenance package and keep the constructor package private?